### ISSUE TITLE
Backups with WAL config not working locally

### ISSUE THREAD

**Issue by humbleCraftman** [thedatabaseme/cookbooks#1]
Hi guys. Were you successful in doing this work on a local k8 cluster set up with kind and minikube. I'm two days away fighting to make it work .Once I have defined the WAL_S3_BUCKET parameter in the configMap, the postgres operator cannot launch the pods.My all cluster gets corrupted and nothing works. Any idea ?
I saw [here](https://github.com/zalando/postgres-operator/blob/master/docs/administrator.md#wal-archiving-and-physical-basebackups) that "When you overwrite the configuration by defining WAL_S3_BUCKET..." can be touchy. How did you manage that ?
Otherwise, great job ! Very helpfull.

**Comment by humbleCraftman** [comment]
**Initial:**
NAME                                    READY   STATUS    RESTARTS      AGE   SPILO-ROLE
postgres-demo-cluster-0                 1/1     Running   0             21m   master
postgres-demo-cluster-1                 1/1     Running   0             18m   replica
postgres-demo-cluster-2                 1/1     Running   0             17m   replica
postgres-demo-cluster-3                 1/1     Running   0             17m   replica

And then I apply kubectl apply -k overlays/enabled-backup/

**After few minutes (pod postgres-demo-cluster-1 not restarting):**
NAME                                    READY   STATUS    RESTARTS      AGE     SPILO-ROLE
postgres-demo-cluster-0                 1/1     Running   0             27m     master
postgres-demo-cluster-1                 0/1     Running   0             5m38s
postgres-demo-cluster-2                 1/1     Running   0             23m     replica
postgres-demo-cluster-3                 1/1     Running   0             23m     replica

**After few more minutes (pod postgres-demo-cluster-1 postgres-demo-cluster-2 and  not restarting, postgres-demo-cluster-3 disappeared):**
NAME                                    READY   STATUS    RESTARTS      AGE   SPILO-ROLE
postgres-demo-cluster-0                 0/1     Running   0             10m
postgres-demo-cluster-1                 1/1     Running   0             41m   master
postgres-demo-cluster-2                 0/1     Running   0             14m

Logs in pods struggling to restart:
2022-06-24 16:42:55.185 40 LOG {ticks: 0, maint: 0, retry: 0}
2022-06-24 16:42:57,441 INFO: no action. I am (postgres-demo-cluster-1), the leader with the lock
2022-06-24 16:43:07,442 INFO: no action. I am (postgres-demo-cluster-1), the leader with the lock
2022-06-24 16:43:17,485 INFO: no action. I am (postgres-demo-cluster-1), the leader with the lock
2022-06-24 16:43:25.216 40 LOG {ticks: 0, maint: 0, retry: 0}
2022-06-24 16:43:27,438 INFO: no action. I am (postgres-demo-cluster-1), the leader with the lock
2022-06-24 16:43:37,438 INFO: no action. I am (postgres-demo-cluster-1), the leader with the lock
2022-06-24 16:43:47,484 INFO: no action. I am (postgres-demo-cluster-1), the leader with the lock
2022-06-24 16:43:55.211 40 LOG {ticks: 0, maint: 0, retry: 0}
2022-06-24 16:43:57,437 INFO: no action. I am (postgres-demo-cluster-1), the leader with the lock
2022-06-24 16:44:07,439 INFO: no action. I am (postgres-demo-cluster-1), the leader with the lock
2022-06-24 16:44:17,484 INFO: no action. I am (postgres-demo-cluster-1), the leader with the lock
2022-06-24 16:44:25.244 40 LOG {ticks: 0, maint: 0, retry: 0}
2022-06-24 16:44:27,437 INFO: no action. I am (postgres-demo-cluster-1), the leader with the lock
2022-06-24 16:44:37,438 INFO: no action. I am (postgres-demo-cluster-1), the leader with the lock

**Comment by thedatabaseme** [comment]
Hello,

I cannot see an error directly from your output. The log output shows only the Patroni status that he's the leader. So maybe I'm confused, but I cannot see the error in your setup. The `enabled-backup` overlay only updates some values in the `pod-config` configmap. Please doublecheck if the parameters in there are updated correctly. Also have you adjusted the parameters to your S3 / Object Storage setup? Else the backup will obviously not work.

Philip

**Comment by humbleCraftman** [comment]
Yes I adjusted the parameters to my S3 (local minio). Just did not paste the good logs with are:

2022-06-24 17:10:07,446 ERROR: Exception when working with leader
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/patroni/postgresql/rewind.py", line 68, in check_leader_is_not_in_recovery
    with get_connection_cursor(connect_timeout=3, options='-c statement_timeout=2000', **conn_kwargs) as cur:
  File "/usr/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/usr/local/lib/python3.6/dist-packages/patroni/postgresql/connection.py", line 44, in get_connection_cursor
    conn = psycopg.connect(**kwargs)
  File "/usr/lib/python3/dist-packages/psycopg2/__init__.py", line 127, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: connection to server at "10.244.0.15", port 5432 failed: session is read-only

2022-06-24 17:10:07,447 INFO: no action. I am (postgres-demo-cluster-0), a secondary, and following a leader (postgres-demo-cluster-1)

**Comment by humbleCraftman** [comment]
Did you use a local minikube/kind cluster to setup this cluster ?

**Comment by thedatabaseme** [comment]
No I didn't use minikube or Kind. But I can test that.

**Comment by thedatabaseme** [comment]
Can you please provide the output of the following commands:

From within one of the db pod (prefered the one that is marked as `leader` if there is any)
```
patronictl list
```

From within the `leader` database pod
```
envdir /run/etc/wal-e.d/env/ wal-g backup-list
```

**Comment by humbleCraftman** [comment]
> No I didn't use minikube or Kind. But I can test that.

In parallel, I will try to make it run in a cloud managed kubernetes cluster..

**Comment by humbleCraftman** [comment]
I found where the error was after things got running in a cloud provided kubernetes cluster rather than locally.
In my local cluster running with minikube and kind as suggested by the zalando postrgres operator [here](https://github.com/zalando/postgres-operator/blob/master/docs/quickstart.md#prerequisites) I set my local minio server as a S3 like object storage. I modified my hosts file to set the url minio.home.lab as localhost. But my k8 cluster is not aware of this setting I my master pod struggles to contact the URL. Making it unhealthy and confusing the other pods to start proprely.
Sorry if it took you some time to investigate.
Thanks for answers.
